src/components/search/SearchSection.jsx

```diff
@@ -104,44 +116,46 @@ const SearchSection = () => {
   };
 
   return (
-    <div id={styles.searchContainer}>
-      <label id={styles.title}>Find Your Next Performer:</label>
+    <SearchContext.Provider value={{ setSearchTerm, setSearchLocation }}>
+      <div id={styles.searchContainer}>
+        <label id={styles.title}>Find Your Next Performer:</label>
 
-      <div id={styles.searchForm}>
-        <div className={styles.searchBar} id={styles.upperSearchBar}>
-          {/* <AutoComplete className={styles.searchInput} suggestions={dummyTags} /> */}
-          <input className={styles.searchInput}
-            onChange={onSearchTermChange}
-            placeholder="Search by event name"
-          />
+        <div id={styles.searchForm}>
+          <div className={styles.searchBar} id={styles.upperSearchBar}>
+            {/* <AutoComplete className={styles.searchInput} suggestions={dummyTags} /> */}
+            <input className={styles.searchInput}
+              onChange={onSearchTermChange}
+              placeholder="Search by event name"
+            />
 
-          <button className={styles.insideBtn}><FaSearch /></button>
-        </div>
-        <div className={styles.searchBar}>
-          <input className={styles.searchInput}
-            onChange={onSearchLocationChange}
-            placeholder="Location"
-          />
-          <button className={styles.insideBtn}><FaMapMarkerAlt /></button>
-        </div>
-        {/* <label>
+            <button className={styles.insideBtn}><FaSearch /></button>
+          </div>
+          <div className={styles.searchBar}>
+            <input className={styles.searchInput}
+              onChange={debounce(onSearchLocationChange, 800)}
```

Comment:
Unless I'm misunderstanding something, I think we don't want to run `onSearchLocationChange` until they've fully entered their address. So it might be better to remove the `debounce` and swap `onChange` (runs with every key press) to `onBlur` (runs when they finish), or even not do it at all until they click the search button.

Reply:
I changed to call geocode when click search button, so only call geocode once for each search.

Reply:
Smart.